src/transport/RendezvousSession.cpp

```diff
@@ -375,13 +380,12 @@ CHIP_ERROR RendezvousSession::HandleSecureMessage(PacketBuffer * msgBuf)
         err = mNetworkProvision.HandleNetworkProvisioningMessage(payloadHeader.GetMessageType(), msgBuf);
         SuccessOrExit(err);
     }
-    // else .. TODO once application dependency on this message has been removed, enable the else condition
+    else // This else condition should eventually be removed, once all messages are handled via delegate callbacks
     {
         mDelegate->OnRendezvousMessageReceived(msgBuf);
```

Comment:
I noticed that RendezvousServer::OnRendezvousMessageReceived is responsible for releasing the buffer so if you're making the call conditional, this function should take care of freeing the buffer.

Reply:
`RendezvousServer::OnRendezvousMessageReceived()` will be removed once nRF apps migrate to delegate based calls. Since delegates are implemented by different pieces of code, I feel it's better to free the buffer here, since this is the common code.

Reply:
Yes, I'm fine with freeing the buffer here. I'm just noticing that we don't do it now. :) We only free origMsg in the epilogue while msgBuf is leaked if I'm not mistaken. But I guess we can address that in another PR.

Reply:
that's a good point. I'll update the PR to take care of it..

Reply:
oops, got merged while I pushed the change. I'll address in a separate PR.